packages/status-core/src/utils/public-key-to-color-hash.ts

```diff
@@ -0,0 +1,85 @@
+import * as secp256k1 from 'ethereum-cryptography/secp256k1'
+
+type ColorHash = number[][]
+
+const COLOR_HASH_COLORS_COUNT = 32
+const COLOR_HASH_SEGMENT_MAX_LENGTH = 5
+
+export function publicKeyToColorHash(publicKey: string): ColorHash | undefined {
+  const publicKeyHex = publicKey.replace(/^0[xX]/, '') // ensures hexadecimal digits without "base prefix"
+
+  let compressedPublicKeyDigits: string
+  try {
+    compressedPublicKeyDigits =
+      secp256k1.Point.fromHex(publicKeyHex).toHex(true) // validates and adds "sign prefix" too
```

Comment:
**TLDR:** I'm not proposing to "compress" the public key even further while preserving the same collision resistance. 

However, I'd really appreciate if you could bear with me on this one and help me reason about why "Status/Waku addresses", if you will, cannot be used in place of public keys for the "visual identity" feature (i.e. cut-offs, color and emoji hashes).

And thus making the resulting hex 20 bytes instead of 33.

So,

1. **Standard-wise**, could the same equation used for Ethereum addresses be used to generate Status addresses? See `A(p_{\mathrm{r}}) = \mathcal{B}_{96..255}\big(\mathtt{KEC}\big( \mathtt{ECDSAPUBKEY}(p_{\mathrm{r}}) \big) \big)`.
![4aae1b2496c4044fb50725832fc82e3f](https://user-images.githubusercontent.com/13265126/169754697-855c8191-3d60-4229-8d7c-2813190dab2b.png)
2. **Security-wise**, if it's enough for Ethereum network wouldn't it be so for Waku too?
3. Does the two work the same in terms of allowed "address space"?
4. By how much would such use be less secure?
5. **Implementation-wise**, the color hash is already loosing some of the resistance. How much?
6. **Usability-wise**, doesn't spotting of the other account(s) rely first only on the color hash. How much collision resistance is in that alone?
7. And how much of the resistance would be in address-based version?
8. Finally, **product-wise** and given the above, would the addresses suffice?

---

To @0kok0 @osmaczko @D4nte @John-44 
Cc @Samyoul @cammellos

Reply:
Interesting concept. I would not really call this a "Waku address" tho as it's Status chat specific.